### PR TITLE
[video_player] Ensures autoplay is false on the web.

### PR DESCRIPTION
**Pull request by ditman**
### Description

This PR changes the initialization of the `video` element used by the web implementation of the video_player plugin to **ensure the `autoplay` attribute is set to _**false**_**, as expected by the code.

### Issues

* Fixes flutter/flutter#135194
* ~~May be the root cause of flutter/flutter#130147~~ (it isn't, I've seen the integration test failing again)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/video_player/video_player_web/lib/src/video_player.dart [hunk 1]
```diff
@@ -66,8 +66,8 @@ class VideoPlayer {
     // Allows Safari iOS to play the video inline
     _videoElement.setAttribute('playsinline', 'true');
 
-    // Set autoplay to false since most browsers won't autoplay a video unless it is muted
-    _videoElement.setAttribute('autoplay', 'false');
+    // Ensure autoplay is not set through the DOM!
```
Comment (stuartmorgan) [inline packages/video_player/video_player_web/lib/src/video_player.dart:69]: Does this mean we should also remove line 63, or is that doing something different? I'm not clear on what exactly the HTML wrapper class properties do.

File: packages/video_player/video_player_web/example/integration_test/video_player_test.dart [hunk 2]
```diff
@@ -31,8 +31,11 @@ void main() {
 
       expect(video.controls, isFalse,
           reason: 'Video is controlled through code');
-      expect(video.getAttribute('autoplay'), 'false',
-          reason: 'Cannot autoplay on the web');
+      expect(video.autoplay, isFalse,
+          reason: 'autoplay attribute on HTMLVideoElement MUST be false');
+      // see: https://developer.mozilla.org/en-US/docs/Glossary/Boolean/HTML
```
Comment (stuartmorgan) [inline packages/video_player/video_player_web/example/integration_test/video_player_test.dart:36]: Ugh, I had forgotten about this.

File: packages/video_player/video_player_web/lib/src/video_player.dart (hunk 1 above)
Comment (ditman) [reply packages/video_player/video_player_web/lib/src/video_player.dart:69]: Let me vanilla-JS this bit, I'm not sure if the DOM attribute is synced with the internal state of the element.

Comment (ditman) [reply packages/video_player/video_player_web/lib/src/video_player.dart:69]: At least in chrome, they're kept in sync (I'm going to assume this is standard behavior), so we could do either:

Screenshot 2023-09-21 at 11 39 42 AM

Comment (ditman) [reply packages/video_player/video_player_web/lib/src/video_player.dart]: I removed the `removeAttribute` call, which is redundant with line 63. Left the check in the tests, which should pass anyway. I also documented why we need to do setAttribute on the "playsinline" property.

File: packages/video_player/video_player_web/example/integration_test/video_player_test.dart (hunk 2 above)
Comment (ditman) [reply packages/video_player/video_player_web/example/integration_test/video_player_test.dart:36]: Yeah, this is a doozy of a feature :)

**Review by stuartmorgan — APPROVED** [review]
LGTM!

**Pull request merged**